extensions/BeatBlox/index.js

```diff
@@ -612,9 +612,10 @@
                             await waitUntil(this.musicInfo.t - SCHEDULING_WINDOW);
                         }, { args: [], timeout: I32_MAX });
                     }),
-                    new Extension.Block('playAudioClipForDuration', 'command', 'music', 'play sound %snd for duration %n secs', [null, 0], function (clip, duration) {
+                    new Extension.Block('playAudioClipForDuration', 'command', 'music', 'play clip %snd for duration %n secs', [null, 0], function (clip, duration) {
                         setupProcess(this);
                         if (clip === '') throw Error(`sound cannot be empty`);
+                        if (duration === 0) throw Error(`duration must be greater than 0`);
```

Comment:
This still lets them put negative durations.

Also, any code you put outside of the `runAsync` is going to execute over and over dozens or hundreds of times, so you might want to move as much as possible inside as long as it doesn't affect the behavior.